File: Binary-Search/binary_search.py
'''
Binary Search

This helps in searching a key in an array
in O(log(N)) time.
'''

arr = [1,2,2,3,6,6,7,9,10]
start = 0
end = len(arr)-1
key = 3

mid = (start+end)//2
found = False

while start<=end:
    if arr[mid]==key:
        found = True
        break
    elif arr[mid]>key:
        end = mid-1
    else:
        start = mid+1
    mid = (start+end)//2

if found:
    print("Key found at index:", mid)
else:
    print("Key not found..")

File: Binary-Search/binary_search.js
<script>
let recursiveFunction = function (arr, x, start, end) {
	

	if (start > end) return false;                					// Base case 


	let mid=Math.floor((start + end)/2);                	// Finds the middle index

	// Compare mid with given key x
	if (arr[mid]===x) return true;
		

	if(arr[mid] > x)
		return recursiveFunction(arr, x, start, mid-1);     	// If element at mid is greater than x,
		                                                    	// search in the left half of mid
	else


		return recursiveFunction(arr, x, mid+1, end);
}                                                          		// If element at middle is smaller than x,
		                                                          // search in the right half of middle

 
let arr = [1, 3, 5, 7, 8, 9];
let x = 5;

if (recursiveFunction(arr, x, 0, arr.length-1))

	document.write("Element found!<br>");
else
 document.write("Element not found<br>");

x = 6;

if (recursiveFunction(arr, x, 0, arr.length-1))
	document.write("Element found!<br>");
else
 document.write("Element not found!<br>");
</script>									
